### PR TITLE
feat (_posts): add web-backend-interview-in-2019

### PR DESCRIPTION
**Pull request by FrankYang0529**
分享一些今年在後端面試的心得與技巧

### CODE REVIEW

File: source/_posts/web-backend-interview-in-2019.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+3. 作業系統（Operating System）
+4. 網路（Network）
+5. 系統設計（System Design）
+6. 特定軟體（像是 MySQL, Redis 等）
+
+以下會針對這六大類，分別介紹並提供一些準備的資源！
+
+## 演算法（Algorithm）
+
+演算法考題就是大家常說的 *[LeetCode](https://leetcode.com)*，我面試的公司中，有的每一關都有類似 LeetCode 的考題，有的則會把這類考題集中在第一關。關於 LeetCode 考題，會建議*別急著刷題，先把演算法的理論與應用理解後，在開始刷題，效率會更好！*
```
Comment (aszx87410) [inline source/_posts/web-backend-interview-in-2019.md]: 在開始刷題 => 再

File: source/_posts/web-backend-interview-in-2019.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+針對理解演算法，會推薦*[極客時間 - 數據結構與算法之美](https://time.geekbang.org/column/intro/126)*，我覺得這是在「有限時間內」準備演算法的好教材！在介紹每個演算法之前，作者都會以一個實際應用的問題開頭，像是介紹 Stack 時，作者會先問「如何實現瀏覽器中上一頁、下一頁的功能？」，然後才介紹 Stack，最後在針對開頭的問題解答，這種學習方式能讓讀者對 Stack 能處理哪類型的問題更有感覺！很多時候面試中的問題並不是直接給 input 和 output，而是給一個應用情境，這時候帶著作者給的問題學習，效果會比較好！
```
Comment (aszx87410) [inline source/_posts/web-backend-interview-in-2019.md]: 最後在針對 => 再

File: source/_posts/web-backend-interview-in-2019.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+針對刷題，我會推薦以下連結：
+
+1. [LeetCode Learn](https://leetcode.com/explore/learn/)，這裡有針對基礎算法整理出來的考題，像是 Queue & Stack, Linked List, Hash Table 等，對於一開始認識題型與認識解題技巧有很大的幫助。
+2. [TechBridge - LeetCode 刷題 Pattern](https://blog.techbridge.cc/tags/Leetcode/)，如果多刷點題，會發現不同的題型常常用到一樣的技巧，這時候認識刷題 Pattern 就很有幫助，透過刷題 Pattern 可以知道哪些類型的問題，都可以用同樣的 Pattern 處理！
+3. [LeetCode Interview](https://leetcode.com/explore/interview/)，最後就是針對想要的公司準備，如果 LeetCode 沒有整理到的，可以直接搜尋「某某公司 面經」，面經中提到的題目雖然不會一直重複出，但面試前先看過以往題目的難度，心裡至少會有個底。
+
+## 程式語言（Programming Language）
+
+程式語言的部分，會建議先看過招聘說明，通常會提到希望面試者會哪些語言，這時候在針對那個語言搜尋「某某語言 面經」，像是 Python 面經、NodeJS 面經等。這部分通常會針對該語言的特性提問，例如 Python 的 Immutable v.s. Mutable 或是 NodeJS Event Loop 等，通常每個語言都會有個經典考題是垃圾回收機制（Garbage Collection），最後要時常關注語言的 Latest Released 有沒有加入哪些新特性，這些也很容易成為面試題目！
```
Comment (aszx87410) [inline source/_posts/web-backend-interview-in-2019.md]: 這時候在針對 => 再

File: source/_posts/web-backend-interview-in-2019.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+## 作業系統（Operating System）
+
+作業系統的經典考題如下：
+
+- 進程（Process） v.s. 線程（Thread） v.s. 協程（Coroutine）
```
Comment (aszx87410) [inline source/_posts/web-backend-interview-in-2019.md]: 可以主要使用臺灣的名詞翻譯（處理程序、執行緒）並在括弧內補上中國的翻譯，或甚至直接用英文也可以。

File: source/_posts/web-backend-interview-in-2019.md (hunk 2 above)
Comment (FrankYang0529) [reply source/_posts/web-backend-interview-in-2019.md]: Fixed it. Thank you!

File: source/_posts/web-backend-interview-in-2019.md (hunk 3 above)
Comment (FrankYang0529) [reply source/_posts/web-backend-interview-in-2019.md]: Fixed it. Thank you!

File: source/_posts/web-backend-interview-in-2019.md (hunk 4 above)
Comment (FrankYang0529) [reply source/_posts/web-backend-interview-in-2019.md]: 好的，這部分我直接使用英文，謝謝。

File: source/_posts/web-backend-interview-in-2019.md (hunk 1 above)
Comment (FrankYang0529) [reply source/_posts/web-backend-interview-in-2019.md]: Fixed it. Thank you!

File: source/_posts/web-backend-interview-in-2019.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+由於大多數學校沒有系統設計相關的課程，所以我會推薦先看*[極客時間 - 從 0 開始學架構](https://time.geekbang.org/column/intro/81)*，這個專欄有點像系統設計的工具箱，介紹不同種類的問題有哪些解決方式，先要知道有哪些工具可以用，這樣看系統設計的問題，才不會覺得是東補一塊西補一塊。
+
+實際演練系統設計相關的面試題，我會推薦 *[Grokking the System Design Interview](https://www.educative.io/courses/grokking-the-system-design-interview)*，這個課程會帶大家設計各種知名系統，並且也會說明實際遇到系統設計的面試題，要怎麼回答比較好。以下是我的一些心得：
+
+1. 跟面試官確定系統需要哪些功能
+2. 評估 DAU（Daily Active User）、流量、QPS / TPS、硬碟儲存空間、記憶體空間等
+3. 設計 API
+4. 設計 Data Model
+5. High Level Desin，大致說明系統需要哪些 Component
```
Comment (Po-Jen) [inline source/_posts/web-backend-interview-in-2019.md]: 應該是 High Level **Design**

**Review by Po-Jen — COMMENTED** [review]
Nice article！ 寫得簡潔又到位！我很喜歡建議先理解演算法和資結的那一段說明，Thanks for sharing 👍

File: source/_posts/web-backend-interview-in-2019.md (hunk 5 above)
Comment (FrankYang0529) [reply source/_posts/web-backend-interview-in-2019.md]: Fixed it. Thank you.

**Comment by kdchang** [comment]
Hi @FrankYang0529,

Nice sharing ! We would merge it later.

BTW, we would gradually migrate to new writing submission mechanism and award system, maybe you can try it next time. thank you :) 

Ref: https://www.coderbridge.com/articles/c63f676ec1ad4e1e8e8bd1f5abb1bcc4

**Review by kdchang — APPROVED** [review]

**Pull request merged**